### PR TITLE
fix: STRF-11923 - Handle Missing a New Channels Permission Requirement in Auth Token

### PR DESCRIPTION
**Pull request by bc-jz**
#### What?
This PR handles the possibility of missing the `Channels` auth permission when making a GET for a list of channels in `getStoreChannels()`. 

When [this update](https://github.com/bigcommerce/stencil-cli/pull/1180) was made I overlooked that the standard Stencil CLI auth token that we recommend providing to 3rd parties does not include the "read channel settings" permission by default. So people utilizing the Stencil CLI token, or any v3 token that does not include "read/manage channel settings" permission, can end up with this error when making `start`, `push`, `pull`, or `download` commands:
Screenshot 2024-04-10 at 7 14 28 AM

We will be updating that permission on the Stencil CLI token but people with errors will still have to generate a new token to get this permission. As such, this PR will avoid failures if the `Channels` permission is missing. Instead we will ignore the error and return an unfiltered list of urls the same way we did prior to the change in version 4.7.2.

**Refactor**
Also incorporates some smart improvements to the fetching of channels that I found in this PR:
https://github.com/bigcommerce/stencil-cli/pull/1187

#### Tickets / Documentation
[Previous related pr.](https://github.com/bigcommerce/stencil-cli/pull/1180) 

#### Screenshots (if appropriate)
With this changes in this PR we will now give a "warning" message if the "channels" permission is missing and still proceed with an unfiltered response as we did before:
Screenshot 2024-04-10 at 7 17 32 AM

However, if you do have an updated token that includes the correct permission then we will filter the list of channels returned using this new information:
Screenshot 2024-04-10 at 7 19 12 AM

cc @bigcommerce/storefront-team

### CODE REVIEW

File: lib/theme-api-client.js [hunk 1]
```diff
@@ -212,19 +212,28 @@ async function getStoreChannels({ accessToken, apiHost, storeHash }) {
...
-        });
+        const channelsResponse = await networkUtils
+            .sendApiRequest({
+                url: `${apiHost}/stores/${storeHash}/v3/channels?type:in=storefront`,
+                accessToken,
+            })
+            .catch((error) => {
+                if (error.response.data.status === 403) {
+                    console.log(
+                        'WARNING: Version 4.7.2+ of stencil-cli added a request to the Channels resource which requires a permission missing on your current stencil cli auth token. Please generate a new auth token at your convenience which includes the "read-only channels" permission to clear up this warning. You can read more about correcting this issue here: https://github.com/bigcommerce/stencil-cli/issues/1185',
```
Comment (jairo-bc) [inline lib/theme-api-client.js:223]: Great idea for the warning!

Comment (jairo-bc) [inline lib/theme-api-client.js:223]:
```suggestion
                        'WARNING: Version 7.4.2+ of stencil-cli added a request to the Channels resource which requires a permission missing on your current stencil cli auth token. Please generate a new auth token at your convenience which includes the "read-only channels" permission to clear up this warning. You can read more about correcting this issue here: https://github.com/bigcommerce/stencil-cli/issues/1185',
```

File: lib/theme-api-client.js [hunk 2]
```diff
@@ -212,19 +212,28 @@ async function getStoreChannels({ accessToken, apiHost, storeHash }) {
...
-            url: `${apiHost}/stores/${storeHash}/v3/channels`,
-            accessToken,
-        });
+        const channelsResponse = await networkUtils
+            .sendApiRequest({
+                url: `${apiHost}/stores/${storeHash}/v3/channels?type:in=storefront`,
+                accessToken,
+            })
+            .catch((error) => {
+                if (error.response.data.status === 403) {
```
Comment (jairo-bc) [inline lib/theme-api-client.js:221]: I think it's 
```suggestion
                if (error.response.status === 403) {
```

Comment (bc-jz) [reply lib/theme-api-client.js:221]: Ahh, its both actually:
```
response: {
    status: 403,
    statusText: 'Forbidden',
    ...
    data: {
      status: 403,
      title: "You don't have a required scope to access the endpoint",
      type: 'https://developer.bigcommerce.com/api-docs/getting-started/api-status-codes',
      errors: {}
    }
  }
```

**Review by jairo-bc — APPROVED** [review]

**Pull request merged**